### PR TITLE
feat(dropdown): added select element to dropdown

### PR DESCRIPTION
**Pull request by hcafaq**
## Pull Request

Added forward ref and HTMLSelectElement props

### Description
Added forward ref and HTMLSelectElement props

Describe your changes (or add ticket description) and add relevant screenshots
Added forward ref and HTMLSelectElement props

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change

### How do I test this

- Add steps to test
- in bullet point format
- preferably you can add link to the storybook build in the PR

### Did you remember to take care of the following?

- [x] `npm i` – for new NPM dependencies.
- [x] `npm run lint` - to check for linting issues
- [x] `npm run test` - to run unit tests
- [x] `npm run test:sh:docker` - to run screenshot tests, [detail instruction](https://hey-car.github.io/heycar-uikit/main/?path=/docs/guidelines-screenshot-testing--page)

### New Feature / Bug Fix

- [ ] Run unit tests to ensure all existing tests are still passing.
- [ ] Add new passing unit tests to cover the code introduced by your pr.

Thanks for contributing!

### CODE REVIEW

**Review by heyabinanto — APPROVED** [review]

File: packages/dropdown/src/Dropdown.tsx [hunk 1]
```diff
@@ -6,104 +6,107 @@ import { DropdownProps, SelectOptions } from './Dropdown.types';
...
+      disabled,
+      dataTestId,
+      onBlur,
+      onClick,
+      fullWidth,
+      ...restProps
+    },
+    ref,
+  ) => {
+    const [stateValue, setStateValue] = useState<SelectOptions | undefined>(
```
Comment (frankie303) [inline packages/dropdown/src/Dropdown.tsx:25]: with something like this, IMO they could be more descriptive:
`stateValue` -> `selectedOption`
`setStateValue` -> `setSelectedOption`

File: packages/dropdown/src/Dropdown.tsx [hunk 2]
```diff
@@ -6,104 +6,107 @@ import { DropdownProps, SelectOptions } from './Dropdown.types';
...
-                </span>
-              )}
-            </div>
-          </li>
-        ))}
-      </ul>
-    </Component>
-  );
-}
+        <span className={valueClassNames}>{stateValue?.label}</span>
```
Comment (frankie303) [inline packages/dropdown/src/Dropdown.tsx:74]: shouldn't we do this check outside of the `span`?

File: packages/dropdown/src/Dropdown.tsx (hunk 1 above)
Comment (hcafaq) [reply packages/dropdown/src/Dropdown.tsx]: resolved

File: packages/dropdown/src/Dropdown.tsx (hunk 2 above)
Comment (hcafaq) [reply packages/dropdown/src/Dropdown.tsx]: resolved

**Review by frankie303 — APPROVED** [review]

File: packages/dropdown/src/Dropdown.tsx [hunk 3]
```diff
@@ -6,104 +6,107 @@ import { DropdownProps, SelectOptions } from './Dropdown.types';
...
+              key={option.value}
+              onClick={e => {
+                e.stopPropagation();
+                selectOption(option);
+                setIsOpen(false);
+              }}
+            >
+              <div className={styles.contentContainer}>
+                {option.leftContent && (
+                  <span className={styles.leftContent}>
```
Comment (esraltintas) [inline packages/dropdown/src/Dropdown.tsx:93]: @hcafaq please check the test coverage

Comment (hcafaq) [reply packages/dropdown/src/Dropdown.tsx:93]: converted to draft will push another commit for it

**Pull request closed without merging**